src/processing.rs

```diff
@@ -79,44 +79,49 @@ pub fn stack_frames(
     );
 
     // Initial necessary values
-    length_signal = sig.shape[0];
-    frame_sample_length = int(np.round(sampling_frequency * frame_length)); // Defined by the number of samples
-    frame_stride = (sampling_frequency * frame_stride).round();
+    let length_signal = sig.shape[0];
+    let frame_sample_length = (sampling_frequency * frame_length).round(); // Defined by the number of samples
+    let frame_stride = (sampling_frequency * frame_stride).round();
+    let mut len_sig = 0;
+    let mut numframes = 0;
 
     // Zero padding is done for allocating space for the last frame.
-    if zero_padding {
+    let signal = if zero_padding {
         // Calculation of number of frames
-        numframes = (int(math.ceil((length_signal - frame_sample_length) / frame_stride)));
-        println!(numframes, length_signal, frame_sample_length, frame_stride);
+        numframes = ((length_signal - frame_sample_length) / frame_stride).ceil() as i32;
+        println!(
+            "{} {} {} {}",
+            numframes, length_signal, frame_sample_length, frame_stride
+        );
 
         // Zero padding
-        len_sig = int(numframes * frame_stride + frame_sample_length);
-        additive_zeros = np.zeros((len_sig - length_signal,));
-        signal = np.concatenate((sig, additive_zeros));
+        len_sig = (numframes * frame_stride + frame_sample_length) as i32;
+        let additive_zeros = ndarray::ArrayBase::zeros((len_sig - length_signal,));
+        ndarray::concatenate(0, &[sig.view(), additive_zeros.view()]);
     } else {
         // No zero padding! The last frame which does not have enough
         // samples(remaining samples <= frame_sample_length), will be dropped!
-        numframes = (length_signal - frame_sample_length).floor() / frame_stride;
+        numframes = (length_signal - frame_sample_length) / frame_stride;
 
         // new length
-        len_sig = int((numframes - 1) * frame_stride + frame_sample_length);
-        signal = sig[0..len_sig];
-    }
+        let len_sig = ((numframes - 1) * frame_stride + frame_sample_length) as i32;
+        sig[0..len_sig]
+    };
 
     // Getting the indices of all frames.
-    let indices = np.tile(np.arange(0, frame_sample_length), (numframes, 1))
-        + np.tile(
-            np.arange(0, numframes * frame_stride, frame_stride),
+    let indices = crate::util::tile(np.arange(0, frame_sample_length), (numframes, 1))
```

Comment:
Having "use crate::util::tile" at the start, and here just "tile" would make it more readable.